lambdas/thumbnail/src/t4_lambda_thumbnail/__init__.py

```diff
@@ -368,8 +368,38 @@ def lambda_handler(request):
         img = format_aicsimage_to_prepped(img)
         # Send to Image object for thumbnail generation and saving to bytes
         img = Image.fromarray(img)
+
         # Generate thumbnail
-        img.thumbnail(size)
+        try:
+            # 'I;16' modes have limited resamplers for scaling (poor quality)
+            # convert to 'I' to use a better resampler.
+            # remove the following two lines to prefer fast (but much lower quality) conversion
+            if img.mode.startswith('I;16'):
+                img = img.convert('I')  # 32-bit (more completely implemented by PIL)
+            # attempt to use the default resampler - we have test images using this.
+            img.thumbnail(size)
+        except ValueError as err:
+            if err.args[0] == 'image has wrong mode':
+                # The default resampler doesn't work with this image mode.
+                # PIL does not support all resamplers with all modes.
+                # These are all of the resamplers available, Ordered highest to lowest quality.
+                fallback_resampler_order = [Image.LANCZOS, Image.BICUBIC, Image.HAMMING,
+                                            Image.BILINEAR, Image.BOX, Image.NEAREST]
+                success = False
+                for resampler in fallback_resampler_order:
+                    try:
+                        img.thumbnail(size, resample=resampler)
+                        success = True
+                        break
+                    except ValueError:
+                        continue
+                if not success:
+                    raise ValueError(
```

Comment:
can we reraise the original here instead? (hopefully with added information that we tried to handle this, but failed).

Reply:
reason being: all PIL errors should look like PIL errors and maybe caller wants to handle so we shouldn't change exception type, etc.

Reply:
> can we reraise the original here instead?

Sure.

>  (hopefully with added information that we tried to handle this, but failed).

As a comment in that area of code where it's re-raised?

Reply:
..or I could attach a message to it, like

```python
except ValuError as err:
    [...]
    err.note = "Re-raised: Tried all known resamplers, but none worked for this image mode."
```
I still think creating a new exception is a more informative path, though, as it'll have both the original traceback and the newly-created one, whereas re-raising only shows the original code path (even though it also was caught, handled, and re-raised).

Reply:
Note that I've peeled this segment of code into a separate function, `generate_thumbnail()` -- the one it was a part of is getting pretty long (70 lines still, after removal of that segment).

Reply:
Go ahead and create a fresh exception. I understand your point of view.